Fixes #11327 | update `requestPointerLock()` additional requirements and update its api (#30450)
mdn/content: files/en-us/web/api/element/requestpointerlock/index.md

A {{jsxref("Promise")}} that resolves with {{jsxref("undefined")}}.

## Security

{{Glossary("Transient activation")}} is required when calling `requestPointerLock()`. The user has to interact with the page or a UI element in order for this feature to work. Also, the target element's associated document must be in the active state.

If calling `requestPointerLock()` with {{domxref("Element.requestFullscreen()", "requestFullscreen()")}}, the `requestPointerLock()` must be called first, because the {{domxref("Element.requestFullscreen()", "requestFullscreen()")}} will consume the state of {{Glossary("Transient activation", "transient activation")}}.

The `allow-pointer-lock` [sandbox token](/en-US/docs/Web/HTML/Element/iframe#sandbox) must be added when calling `requestPointerLock()` in an {{htmlelement("iframe")}} element. Also, no other elements in other {{htmlelement("iframe")}} elements may be in pointer lock mode.

> 

Reviewer: Is it worth mentioning?

> A [requestPointerLock](https://w3c.github.io/pointerlock/#dom-element-requestpointerlock)() call immediately after the [default unlock gesture](https://w3c.github.io/pointerlock/#dfn-default-unlock-gesture) MUST fail even when [transient activation](https://html.spec.whatwg.org/multipage/interaction.html#transient-activation) is available, to prevent malicious sites from acquiring an unescapable locked state through repeated lock attempts. On the other hand, a [requestPointerLock](https://w3c.github.io/pointerlock/#dom-element-requestpointerlock)() call immediately after a programmatic lock exit (through a [exitPointerLock](https://w3c.github.io/pointerlock/#dom-document-exitpointerlock)() call) MUST succeed when [transient activation](https://html.spec.whatwg.org/multipage/interaction.html#transient-activation) is available, to enable applications to move frequently between interaction modes, possibly through a timer or remote network activity.

Author: good idea!

> When calling `requestPointerLock()` in an {{htmlelement("iframe")}} element, the `allow-pointer-lock` [sandbox token](/en-US/docs/Web/HTML/Element/iframe#sandbox) is required to add. Also, it is required that there is no other element in other {{htmlelement("iframe")}} elements is in pointer lock mode.

Reviewer: ```suggestion
The `allow-pointer-lock` [sandbox token](/en-US/docs/Web/HTML/Element/iframe#sandbox) must be added when calling `requestPointerLock()` in an {{htmlelement("iframe")}} element. Also, no other elements in other {{htmlelement("iframe")}} elements may be in pointer lock mode.
```

A {{jsxref("Promise")}} that resolves with {{jsxref("undefined")}}.

## Security

{{Glossary("Transient activation")}} is required when calling `requestPointerLock()`. The user has to interact with the page or a UI element in order for this feature to work. Also, the target element's associated document must be in the active state.

> If calling `requestPointerLock()` immediately after releasing the pointer lock via default unlock gesture instead of through a `exitPointerLock()` call, even a {{Glossary("transient activation")}} is available, the call will fail.

Reviewer: ```suggestion
If calling `requestPointerLock()` immediately after releasing the pointer lock via the default unlock gesture (instead of through a `exitPointerLock()` call), the call will fail even a {{Glossary("transient activation")}} is available.
```
